lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
@peterhuene @MikaelSmith thoughts about adding a core fact? It currently lives in https://github.com/xaque208/puppet-pkgng/blob/master/lib/facter/pkgng.rb#L22-L38. Other options?